File: twoSums/twoSums.py
"""
Problem Name: Two Sum
Difficulty: Easy
Runtime: 40ms (beats 98.14% of submissions)
Memory: 15MB (beats 70.79% of submissions)
"""


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        val_dict = {}
        result = []
        for idx, val in enumerate(nums):
            rem = target - val
            if rem in val_dict:
                result.append(idx)
                result.append(val_dict[rem])
            val_dict[val] = idx
        return result

File: twoSums/twoSums.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> val_map = new HashMap<Integer, Integer>();
        int result[] = new int[2];
        for(int i=0; i < nums.length; i++) {
            int rem = target - nums[i];
            if (val_map.containsKey(rem)){
                result[0] = val_map.get(rem);
                result[1] = i;
            }
            val_map.put(nums[i], i);
        }
        return result;
    }
}
